File: python-cli/src/convex_notes/handlers/create.py
"""Create note handler."""

from rich.console import Console
from rich.prompt import Prompt

from convex_notes.service import NoteService

console = Console()


def execute(service: NoteService) -> None:
    """Execute the create note command."""
    console.print("\n[bold green]✏️ Create a new note[/bold green]")

    title = Prompt.ask("[cyan]Title[/cyan]")
    content = Prompt.ask("[cyan]Content[/cyan]")

    if not title.strip() or not content.strip():
        console.print("[red]❌ Title and content are required.[/red]")
        return

    try:
        note_id = service.create(title, content)
        console.print(f"\n[green]✅[/green] Note '[yellow]{title}[/yellow]' created!")
        console.print(f"   [dim]ID:[/dim] {note_id}")
    except Exception as e:
        console.print(f"[red]❌ Failed to create note:[/red] {e}")

File: go-cli/internal/handler/create.go
// Package handler contains CLI command handlers.
package handler

import (
	"fmt"

	"convex-notes-cli/internal/service"

	"github.com/AlecAivazis/survey/v2"
	"github.com/fatih/color"
)

// CreateHandler handles the create note command.
type CreateHandler struct {
	service *service.NoteService
}

// NewCreateHandler creates a new CreateHandler.
func NewCreateHandler(svc *service.NoteService) *CreateHandler {
	return &CreateHandler{service: svc}
}

// Execute runs the handler.
func (h *CreateHandler) Execute() error {
	green := color.New(color.FgGreen, color.Bold)
	greenNormal := color.New(color.FgGreen)
	yellow := color.New(color.FgYellow)
	red := color.New(color.FgRed)

	green.Println("\n✏️ Create a new note")

	var title, content string

	if err := survey.AskOne(&survey.Input{Message: "Title:"}, &title); err != nil {
		return err
	}

	if title == "" {
		red.Println("❌ Title is required")
		return nil
	}

	if err := survey.AskOne(&survey.Input{Message: "Content:"}, &content); err != nil {
		return err
	}

	if content == "" {
		red.Println("❌ Content is required")
		return nil
	}

	id, err := h.service.Create(title, content)
	if err != nil {
		return fmt.Errorf("failed to create note: %w", err)
	}

	greenNormal.Printf("\n✅ Note '%s' created!\n", yellow.Sprint(title))
	fmt.Printf("   ID: %s\n", id)

	return nil
}
